sky/skylet/providers/kubernetes/node_provider.py

```diff
@@ -209,10 +210,42 @@ def create_node(self, node_config, tags, count):
         start = time.time()
         while True:
             if time.time() - start > self.timeout:
-                raise config.KubernetesError(
+                pod_status = new_nodes[0].status.phase
+                pod_name = new_nodes[0]._metadata._name
+                events = kubernetes.core_api().list_namespaced_event(
+                    self.namespace,
+                    field_selector=f"involvedObject.name={pod_name}")
+                for event in events.items:
+                    if event.reason == 'FailedScheduling':
+                        event_message = event.message
+                        break
+                timeout_err_msg = (
                     'Timed out while waiting for nodes to start. '
                     'Cluster may be out of resources or '
                     'may be too slow to autoscale.')
+                lack_resource_msg = (
+                    'More than available {0} are requested. '
+                    'Run \'sky status\' to see the amount of {0} '
+                    'that are already in use.')
```

Comment:
`sky status` does not show amount of resources already in use, we should perhaps remove it.
```suggestion
                    'Insufficient {resource}. Other SkyPilot tasks or pods on the cluster may be using resources. Check resource usage by running `kubectl describe nodes`.'
```